proxy/certs/certs.go

```diff
@@ -249,6 +265,23 @@ func parseCert(pemCert string) (*x509.Certificate, error) {
 	return x509.ParseCertificate(bl.Bytes)
 }
 
+// Return the RSA private key, which is lazily initialized.
+func (s *RemoteCertSource) generateKey() *rsa.PrivateKey {
+	s.mutex.Lock()
+	defer s.mutex.Unlock()
+	if s.key != nil {
+		return s.key
+	}
+	start := time.Now()
+	pkey, err := rsa.GenerateKey(rand.Reader, 2048)
+	if err != nil {
+		panic(err) // very unexpected.
+	}
+	logging.Verbosef("Generated RSA key in %v", time.Since(start))
+	s.key = pkey
```

Comment:
I think this can be optimized with a [`Once`](https://pkg.go.dev/sync#Once.Do) instead of a mutex

Reply:
Done, yeah that's a bit simpler.